### DISCUSSION TITLE
Different Debounce Different Parts of Layout

### DISCUSSION THREAD

**Discussion by sofia-bg** [UNC-Project-Open-AAC/OS-DPI#147] [Help]
Design aim: A grid stacked on a button where the button has a shorter debounce than the grid. I was thinking I could do this by adding conditions to the method (in this case with pointer dwell). Alternatively, would there be a way to assign methods/cues/patterns to different parts of a stack? 

Design saved [here](https://github.com/UNC-Project-Open-AAC/OS-DPI/blob/main/Designs%20In%20Progress/IDD01%20Access%20Solutions/Clear_Dwell_Different/IDD01_Clear_Button_Dwell_v1.osdpi). 

![Different_Dwell](https://user-images.githubusercontent.com/92817124/211395510-4c74e5b5-2153-4d46-bf7c-9d47753764eb.png)

**Comment by gbishop** [comment]
Do you mean debounce or dwell? Debounce happens at the lowest level to condition them before your Method sees them at all. Dwell has to be implemented by your method using a timer as in the pointer dwell example. 

I *think* you should be able to do either though I never considered that possibility. It seems like you should be able to test the #ComponentName field to see what component you are in and effectively have different methods active in different components. 

You can easily have different patterns for different components. And your patterns and methods can explicitly name the cues to use if you wish.

**Comment by sofia-bg** [comment]
I meant dwell but for some reason had it in my head that you were using debounce instead of dwell. Obviously haven't spent much time in access tabs since they were added. 

And I see this is the same as where I was stuck before, I thought it wasn't letting me name conditions but I need to go to Add to add the condition. Will work on this tomorrow.

**Comment by sofia-bg** [comment]
[This solution works](https://github.com/UNC-Project-Open-AAC/OS-DPI/blob/main/Designs%20In%20Progress/IDD01%20Access%20Solutions/Clear_Dwell_Different/IDD01_Clear_Button_Dwell_v2.osdpi)! The dwell on the grid is set to 1.5 seconds and cue = yellow. Dwell on clear button is .5 seconds and cue = red. 

Chose those dwells simply hoping it is sufficiently different to show the difference, could exaggerate more.

**Comment by gbishop** [comment]
That is a bug in the Button component. It should be setting the origin to
the name but it fails to do that.

**Reply by gbishop** [reply]
I was confused above. Button did not need an origin field.

**Comment by sofia-bg** [comment]
[Here](https://github.com/UNC-Project-Open-AAC/OS-DPI/blob/main/Designs%20In%20Progress/IDD01%20Access%20Solutions/Clear_Dwell_Different/IDD01_Clear_Button_Dwell_v3.osdpi) is another solution, with a clear button on the left and undo on the right. 

Everything is working ok with the point and dwell, but when I click either nothing happens in display or the label is entered twice. Neither clearing state nor refreshing browser fixing this

**Reply by sofia-bg** [reply]
probably because I got rid of mouse? need to turn attention to other work but will see if adding it back solves the problem

**Comment by gbishop** [comment]
I got confused by your use of origin. You are getting it from the sheet.
That blew my mind for a minute because I use origin in actions...

I suggest using #ComponentName in your rules. That will pick up the name of
the grid or button or whatever and you won't have to add an origin column
to your sheet.

gb
